module/irrecoverable/example_test.go

```diff
@@ -0,0 +1,136 @@
+package irrecoverable_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/lifecycle"
+)
+
+type TemporaryError struct {
+	message string
+}
+
+func (s TemporaryError) Error() string {
+	return s.message
+}
+
+func Example() {
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I'd love to see a comment indicating the context is mandatory, and how we thread the irrecoverable signaling. "This doesn't work w/o a context" should be spectacularly clear.